plutus-benchmark/flat/Codec.hs

```diff
@@ -0,0 +1,71 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE PackageImports    #-}
+
+module Codec ( Codec(..)
+             , flatCodec
+             , cborCodec
+             , withZlib
+             , withPureZlib
+             , codecs
+             ) where
+
+import           Control.DeepSeq                     (NFData, force)
+import qualified Data.ByteString                     as BS
+import qualified Data.ByteString.Lazy                as LBS
+import           Data.Either                         (fromRight)
+import           Data.Text                           (Text)
+
+import           Language.PlutusCore                 (Name (..), Unique (..))
+import           Language.PlutusCore.Universe
+import           Language.UntypedPlutusCore
+import           Language.UntypedPlutusCore.DeBruijn
+
+import           "pure-zlib" Codec.Compression.Zlib  as PureZlib
+import           "zlib" Codec.Compression.Zlib       as Zlib
+import           Codec.Serialise                     (deserialise, serialise)
+import           Flat                                (DecodeException (..), flat, unflat)
+
+type Tm = Term Name DefaultUni ()
+
+data Codec a = Codec
+  { serialize   :: a -> BS.ByteString
+  , deserialize :: BS.ByteString -> a
+  }
+
+fromDecoded :: Show error => Either error a -> a
+fromDecoded (Left err) = error $ show err
+fromDecoded (Right  v) = v
+
+flatCodec :: Codec (Tm)
+flatCodec = Codec
+  { serialize   = force . flat
```

Comment:
I don't know if we need all these `force`s here.  I _think_ that when you use `nf` or `whnf` in Criterion it forces the first argument for you so that you're only benchmarking the time of the application. I'm not at all confident about this stuff though, so it's probably better to be on the safe side and keep the `force`s.

Reply:
My thoughts, exactly. :)